### DISCUSSION TITLE
Can I use CsWin32 to load COM interfaces?

### DISCUSSION THREAD

**Discussion by Slion** [microsoft/CsWin32#504] [Q&A]
Can I use CsWin32 to load COM interfaces?
I'm thinking of using it to load COM interfaces from the Virtual Desktop Manager.
Are there any sample I could take a look at?

**Comment by AArnott** [comment]
CsWin32 reads metadata from the https://github.com/microsoft/win32metadata repo. If the COM interface you're looking for is defined there, then yes you can get the interface from it.

We're adding supported for reading other metadata files that follow the same format and convention as the one produced by that repo, so theoretically you might be able to generate interfaces from other metadata libraries, but this isn't something I have personally tried except for other metadata produced by the same team as the one in win32metadata.
